core/src/main/java/org/apache/brooklyn/core/OsgiLauncher.java

```diff
@@ -0,0 +1,557 @@
+////            // We created the management context, so we are responsible for terminating it
+////            BrooklynShutdownHooks.invokeTerminateOnShutdown(managementContext);
+//    }
+//    /**
+//     * @return The canonical path of the argument.
+//     */
+//    private File resolveSymbolicLink(File f) {
+//        File f2 = f;
+//        try {
+//            f2 = f.getCanonicalFile();
+//            if (Files.isSymbolicLink(f.toPath())) {
+//                log.debug("Resolved symbolic link: {} -> {}", f, f2);
+//            }
+//        } catch (IOException e) {
+//            log.warn("Could not determine canonical name of file " + f + "; returning original file", e);
+//        }
+//        return f2;
+//    }
+//
+//    private void checkFileReadable(File f) {
+//        if (!f.exists()) {
+//            throw new FatalRuntimeException("File " + f + " does not exist");
+//        }
+//        if (!f.isFile()) {
+//            throw new FatalRuntimeException(f + " is not a file");
+//        }
+//        if (!f.canRead()) {
+//            throw new FatalRuntimeException(f + " is not readable");
+//        }
+//    }
+//
+//    private void checkFilePermissionsX00(File f) {
+//
+//        Maybe<String> permission = FileUtil.getFilePermissions(f);
+//        if (permission.isAbsent()) {
+//            log.debug("Could not determine permissions of file; assuming ok: " + f);
+//        } else {
+//            if (!permission.get().subSequence(4, 10).equals("------")) {
+//                throw new FatalRuntimeException("Invalid permissions for file " + f + "; expected ?00 but was " + permission.get());
+//            }
+//        }
+//    }
+
+    /* equivalent of launcher.start() in Main.java */
+    private void launcherStart() {
+        // not needed: management context already running (courtesy of blueprint.xml)
+//        // Create the management context
+//        initManagementContext();
+
+        // Inform catalog initialization that it is starting up
+        CatalogInitialization catInit = ((ManagementContextInternal) managementContext).getCatalogInitialization();
+        catInit.setStartingUp(true);
+
+        // not needed: webapps are started as different features (brooklyn-rest-resources-cxf, brooklyn-jsgui etc)
+//        // Start webapps as soon as mgmt context available -- can use them to detect progress of other processes
+//        if (startWebApps) {
+//            try {
+//                startWebApps();
+//            } catch (Exception e) {
+//                handleSubsystemStartupError(ignoreWebErrors, "core web apps", e);
+//            }
+//        }
+        // not needed: camp is a separate karaf feature - let it initialize itself
+//        // Add a CAMP platform
+//        campPlatform = new BrooklynCampPlatformLauncherNoServer()
+//                .useManagementContext(managementContext)
+//                .launch()
+//                .getCampPlatform();
```

Comment:
Well, the camp bundles could simply bind to the management context service provided by `brooklyn-core` and initialize themselves. This would enable separate deployment of the `brooklyn-core` and `brooklyn-camp` features. If we initialize camp here we need to tie them closer, so basically `brooklyn-core` won't work without the `camp` feature on the classpath (instead of being decoupled through microservice calls).

@neykov I don't know which is best. What is your take on this?